tests/sentry/incidents/test_action_handlers.py

```diff
@@ -302,3 +301,119 @@ class MsTeamsActionHandlerFireTest(MsTeamsActionHandlerBaseTest, TestCase):
+            service_name=SERVICES[0]["service_name"],
+            integration_key=SERVICES[0]["integration_key"],
+            organization_integration=self.integration.organizationintegration_set.first(),
+        )
+
+    def test_build_incident_attachment(self):
+        from sentry.integrations.pagerduty.utils import build_incident_attachment
+
+        alert_rule = self.create_alert_rule()
+        incident = self.create_incident(alert_rule=alert_rule)
+        update_incident_status(
+            incident, IncidentStatus.CRITICAL, status_method=IncidentStatusMethod.RULE_TRIGGERED
+        )
+        integration_key = "pfc73e8cb4s44d519f3d63d45b5q77g9"
+        metric_value = 1000
+        data = build_incident_attachment(incident, integration_key, metric_value)
+
+        assert data["routing_key"] == "pfc73e8cb4s44d519f3d63d45b5q77g9"
+        assert data["event_action"] == "trigger"
+        assert data["dedup_key"] == "incident_{}_{}".format(
+            incident.organization_id, incident.identifier
+        )
+        assert (
+            data["payload"]["summary"] == "1000 events in the last 10 minutes\nFilter: level:error"
+        )
+        assert data["payload"]["severity"] == "critical"
+        assert data["payload"]["source"] == incident.identifier
+        assert data["payload"]["custom_details"] == "Sentry Incident | {}".format(
+            incident.date_started.strftime("%b %d")
+        )
+        assert data["links"][0]["text"] == "Critical: {}".format(alert_rule.name)
+        assert data["links"][0]["href"] == "http://testserver/organizations/baz/alerts/1/"
+
+    @responses.activate
+    def run_test(self, incident, method):
+        from sentry.integrations.pagerduty.utils import build_incident_attachment
+
+        action = self.create_alert_rule_trigger_action(
+            target_identifier=self.service.id,
+            type=AlertRuleTriggerAction.Type.PAGERDUTY,
+            target_type=AlertRuleTriggerAction.TargetType.SPECIFIC,
+            integration=self.integration,
+        )
+
+        responses.add(
+            method=responses.POST,
+            url="https://events.pagerduty.com/v2/enqueue/",
+            body={},
+            status=202,
+            content_type="application/json",
+        )
+        handler = PagerDutyActionHandler(action, incident, self.project)
+        metric_value = 1000
+        with self.tasks():
+            getattr(handler, method)(metric_value)
+        data = responses.calls[0].request.body
+
+        assert json.loads(data) == build_incident_attachment(
+            incident, self.service.integration_key, metric_value
+        )
+
+
+class PagerDutyActionHandlerFireTest(PagerDutyActionHandlerBaseTest, TestCase):
+    def test_fire_metric_alert(self):
+        alert_rule = self.create_alert_rule()
+        self.run_test(self.create_incident(status=2, alert_rule=alert_rule), "fire")
+
+    def test_fire_metric_alert_multiple_services(self):
+        SERVICES = [
```

Comment:
@ceorourke Nit: since you are only making one service at a time, no reason to make an array called `SERVICES`. Also, usually an all caps variable should be for a constant and not a variable.